src/sagas/wallet.js

```diff
@@ -162,7 +162,10 @@ export function* startWallet(action) {
   // but the wallet may be closed unexpectedly
   const storage = STORE.getStorage();
   yield call([storage.store, storage.store.cleanMetadata]); // clean metadata on memory
-  yield call([storage, storage.cleanStorage], true); // clean transaction history
+  // clean transaction history, addresses, token and nano-contract indexes
+  // XXX: It would be better to keep registered tokens and nano contracts
+  // if the network has same genesis
+  yield call([storage, storage.cleanStorage], true, true, true);
```

Comment:
I have two questions:

1. When is this method called? If I open the app the wallet locked (but started) and put my PIN, would it call this method to start my loaded wallet? In this case it would clean the token data previously added, is that correct?

2. Are you cleaning the registered tokens and nano contracts from redux as well?

Reply:
1. You caught a nasty bug :bug: I reverted the changes on https://github.com/HathorNetwork/hathor-wallet-mobile/pull/660/commits/a0a0ea1babe261b7c4f806c1eee3ec4d7092048f and implemented the correct fix on https://github.com/HathorNetwork/hathor-wallet-mobile/pull/660/commits/dd01b71ce759076a458aa51023ec55070a4a6955 using the same method, but on the correct place: the `networkSettings` saga.

2. The registered tokens are being cleaned by the solution described above, but the nano contracts are cleaned on wallet reload. See [the reducer code here](https://github.com/HathorNetwork/hathor-wallet-mobile/blob/77626f8ad8cd3208cb1de24428f89cf43693cc3f/src/reducers/reducer.js#L1442-L1448).